driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncMongoCursor.java

```diff
@@ -51,6 +61,12 @@ class SyncMongoCursor<T> implements MongoCursor<T> {
     SyncMongoCursor(final Publisher<T> publisher, @Nullable final Integer batchSize) {
         this.batchSize = batchSize;
         CountDownLatch latch = new CountDownLatch(1);
+
+        if (isWaitForBatchCursorCreationEnabled()) {
```

Comment:
@vbabanin can you suggest some inline documentation here that describes how it's achieving the intent?

Reply:
I would suggest adding this comment for clarity:

  /*
     * This hook allows us to intercept the `onNext` and `onError` signals for any operation to determine
     * if the {@link BatchCursor} was created successfully, or if an error occurred during its creation process.
     * {@link CompletableFuture} is used to signal when an error is intercepted or when a {@link BatchCursor}
     * is received, we block until it is completed.
     */